File: gear-ecommerce-back/src/main/java/tht/closure/operator/model/entity/UserAddress.java
package tht.closure.operator.model.entity;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

@Entity(name = "t_user_address")
@Getter
@Setter
public class UserAddress extends AbstractEntity {

    private String name;

    private String email;

    private String phone;

    private String street;

    @ManyToOne
    @JoinColumn
    private VnGeographyProvince province;

    @ManyToOne
    @JoinColumn
    private VnGeographyDistrict district;

    @ManyToOne
    @JoinColumn
    private VnGeographyWard ward;

    @OneToOne
    @MapsId
    @JoinColumn
    private User user;
}

File: gear-ecommerce-front/src/material/model/UserAddress.ts
import {AbstractModel} from "./main/AbstractModel";
import {User} from "./User";
import {VnGeographyProvince} from "./VnGeographyProvince";
import {VnGeographyDistrict} from "./VnGeographyDistrict";
import {VnGeographyWard} from "./VnGeographyWard";

export interface UserAddress extends AbstractModel {
    user?: User | undefined;
    name: string | undefined;
    email: string | undefined;
    phone?: string | undefined;
    street?: string | undefined;
    province?: VnGeographyProvince | undefined;
    district?: VnGeographyDistrict | undefined;
    ward?: VnGeographyWard| undefined;
}
